Documentation/Design/Classes/Member Variables.md

```diff
@@ -0,0 +1,277 @@
+# Member Variable Retrieval
+
+## Feature Overview
+
+Classes may define member variables that each instance of that class contains.
+Class instances may retrieve those variables.
+
+Many languages, such as C# and Java, guarantee class instances are created with space for each member.
+More dynamic languages may declare members without guaranteeing their existence (TypeScript).
+Others such as Python and Ruby forgo declaring them altogether under certain conditions.
+
+
+## Commands
+
+### `member variable`
+
+`member variable` `:` *`[privacy]`* `instanceName` `variableName`
+
+Retrieving a member variable will be done with the `member variable` command.
+This takes in the variable's privacy optionally, then the instance name and variable name.
+
+Privacy may be `"public"`, `"protected"`, or `"private"`, and defaults to `public"` if not provided.
+For languages that don't support privacy, variable name name may receive a prefix as per the language's convention.
+
+### `member variable declare`
+
+`member variable declare` `:` *`[privacy]`* `name` `type`
+
+Declaring a member variable will be done with the `member variable declare` command.
+This takes in the variable's privacy, name, and type.
+
+Privacy may be `"public"`, `"protected"`, or `"private"`, and defaults to `public"` if not provided.
+For languages that don't support privacy, variable name name may receive a prefix as per the language's convention.
+
+Some languages don't declare member variables in some or all circumstances. These will consider the `member variable declare` command a no-op.
+
+
+## Usage
+
+```gls
+member variable declare : public name string
+member variable declare : protected age int
+member variable declare : private gender string
+```
+
+```gls
+member variable : name person
+member variable : public name person
+member variable : protected age { this }
```

Comment:
Are these usage examples backwards? 
In the doc it says `member variable` `:` *`[privacy]`* `instanceName` `variableName` 
Wouldn't `instanceName` be `{ this }`/`person`?
I like that order better, because 
`member variable: { this } name`
is closer to this.name than
`member variable: name { this }`